File: scripts/generate_og_image.py
import os
from PIL import Image, ImageDraw, ImageFont

def generate_og_image():
    # Dimensions: 1200 x 630 (Standard Open Graph Banner)
    width, height = 1200, 630
    
    # Create dark mahogany gradient image
    img = Image.new('RGBA', (width, height), (31, 16, 9, 255))
    draw = ImageDraw.Draw(img)
    
    # Background radial glow effect
    for r in range(450, 0, -5):
        alpha = int(25 * (1 - r / 450))
        draw.ellipse([600 - r * 1.5, 315 - r, 600 + r * 1.5, 315 + r], fill=(84, 49, 24, alpha))
        
    # Outer decorative gold border
    draw.rectangle([30, 30, width - 30, height - 30], outline=(217, 130, 91, 180), width=3)
    draw.rectangle([40, 40, width - 40, height - 40], outline=(247, 235, 220, 80), width=1)
    
    # Draw Left Text Content
    # Title "RadhaCafe"
    try:
        font_title = ImageFont.truetype("georgia.ttf", 84)
        font_badge = ImageFont.truetype("arial.ttf", 22)
        font_sub = ImageFont.truetype("arial.ttf", 28)
        font_feat = ImageFont.truetype("arial.ttf", 20)
    except Exception:
        font_title = font_badge = font_sub = font_feat = ImageFont.load_default()

    # Category Badge
    draw.rounded_rectangle([90, 85, 390, 130], radius=22, fill=(217, 130, 91, 45), outline=(217, 130, 91, 120), width=2)
    draw.text((115, 96), "☕ ARTISANAL CAFE & POS", fill=(217, 130, 91, 255), font=font_badge)
    
    # Brand Name
    draw.text((90, 160), "Radha", fill=(247, 235, 220, 255), font=font_title)
    # Calculate offset for "Cafe"
    title_bbox = draw.textbbox((90, 160), "Radha", font=font_title)
    draw.text((title_bbox[2] + 4, 160), "Cafe", fill=(217, 130, 91, 255), font=font_title)
    
    # Subtitle
    draw.text((90, 270), "Modern Cafe Management, Billing,", fill=(247, 235, 220, 230), font=font_sub)
    draw.text((90, 310), "Real-time Analytics & Thermal Printing", fill=(247, 235, 220, 230), font=font_sub)
    
    # Feature Pills
    features = ["⚡ Digital Billing", "📊 Live Analytics", "🖨️ Thermal Receipts"]
    fx = 90
    for feat in features:
        fbox = draw.textbbox((fx, 400), feat, font=font_feat)
        fw = fbox[2] - fbox[0] + 30
        draw.rounded_rectangle([fx, 390, fx + fw, 435], radius=10, fill=(255, 255, 255, 18), outline=(255, 255, 255, 40), width=1)
        draw.text((fx + 15, 400), feat, fill=(247, 235, 220, 255), font=font_feat)
        fx += fw + 16

    # Right side: Circular Vintage Emblem
    cx, cy, radius = 940, 315, 190
    draw.ellipse([cx - radius, cy - radius, cx + radius, cy + radius], fill=(247, 235, 220, 255))
    draw.ellipse([cx - radius + 8, cy - radius + 8, cx + radius - 8, cy + radius - 8], fill=(31, 16, 9, 255))
    draw.ellipse([cx - radius + 18, cy - radius + 18, cx + radius - 18, cy + radius - 18], outline=(247, 235, 220, 255), width=6)
    draw.ellipse([cx - radius + 28, cy - radius + 28, cx + radius - 28, cy + radius - 28], outline=(217, 130, 91, 255), width=3)
    
    # Inner medallion
    draw.ellipse([cx - 105, cy - 105, cx + 105, cy + 105], fill=(84, 49, 24, 255))
    draw.ellipse([cx - 100, cy - 100, cx + 100, cy + 100], outline=(247, 235, 220, 255), width=3)
    
    # Cup body & liquid inside medallion
    draw.polygon([(cx - 45, cy - 35), (cx - 35, cy + 35), (cx + 35, cy + 35), (cx + 45, cy - 35)], fill=(247, 235, 220, 255))
    draw.ellipse([cx - 45, cy - 43, cx + 45, cy - 27], fill=(247, 235, 220, 255), outline=(31, 16, 9, 255), width=2)
    draw.ellipse([cx - 38, cy - 41, cx + 38, cy - 29], fill=(31, 16, 9, 255))
    # Handle
    draw.arc([cx + 30, cy - 25, cx + 65, cy + 25], start=-90, end=90, fill=(247, 235, 220, 255), width=8)
    # Saucer
    draw.ellipse([cx - 60, cy + 33, cx + 60, cy + 47], fill=(247, 235, 220, 255), outline=(31, 16, 9, 255), width=2)
    
    # Save og-image.png
    os.makedirs('public', exist_ok=True)
    og_path = os.path.join('public', 'og-image.png')
    img.save(og_path, 'PNG')
    print(f"Successfully generated {og_path} (1200x630)")
    
    # Generate 512x512 Square Logo Image
    logo_size = 512
    logo_img = Image.new('RGBA', (logo_size, logo_size), (31, 16, 9, 255))
    logo_draw = ImageDraw.Draw(logo_img)
    
    lcx, lcy, lr = 256, 256, 230
    logo_draw.ellipse([lcx - lr, lcy - lr, lcx + lr, lcy + lr], fill=(247, 235, 220, 255))
    logo_draw.ellipse([lcx - lr + 10, lcy - lr + 10, lcx + lr - 10, lcy + lr - 10], fill=(31, 16, 9, 255))
    logo_draw.ellipse([lcx - lr + 22, lcy - lr + 22, lcx + lr - 22, lcy + lr - 22], outline=(247, 235, 220, 255), width=8)
    logo_draw.ellipse([lcx - lr + 34, lcy - lr + 34, lcx + lr - 34, lcy + lr - 34], outline=(217, 130, 91, 255), width=4)
    
    logo_draw.ellipse([lcx - 130, lcy - 130, lcx + 130, lcy + 130], fill=(84, 49, 24, 255))
    logo_draw.ellipse([lcx - 124, lcy - 124, lcx + 124, lcy + 124], outline=(247, 235, 220, 255), width=4)
    
    logo_draw.polygon([(lcx - 60, lcy - 45), (lcx - 45, lcy + 45), (lcx + 45, lcy + 45), (lcx + 60, lcy - 45)], fill=(247, 235, 220, 255))
    logo_draw.ellipse([lcx - 60, lcy - 55, lcx + 60, lcy - 35], fill=(247, 235, 220, 255), outline=(31, 16, 9, 255), width=3)
    logo_draw.ellipse([lcx - 50, lcy - 52, lcx + 50, lcy - 38], fill=(31, 16, 9, 255))
    logo_draw.arc([lcx + 40, lcy - 35, lcx + 85, lcy + 35], start=-90, end=90, fill=(247, 235, 220, 255), width=10)
    logo_draw.ellipse([lcx - 80, lcy + 42, lcx + 80, lcy + 60], fill=(247, 235, 220, 255), outline=(31, 16, 9, 255), width=3)

    for font_file, logo_out in [
        ('logo.png', (512, 512)),
        ('apple-touch-icon.png', (180, 180)),
        ('favicon.png', (64, 64))
    ]:
        out_p = os.path.join('public', font_file)
        resized = logo_img.resize(logo_out, Image.Resampling.LANCZOS)
        resized.save(out_p, 'PNG')
        print(f"Successfully generated {out_p} ({logo_out[0]}x{logo_out[1]})")

if __name__ == '__main__':
    generate_og_image()

File: scripts/generate-og-image.mjs
import { chromium } from 'playwright';
import fs from 'fs';
import path from 'path';

async function generateImages() {
  const browser = await chromium.launch();
  const page = await browser.newPage();

  // SVG Emblem string
  const svgContent = `
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" style="width:100%; height:100%;">
      <defs>
        <path id="rc-top-arc" d="M 95,256 A 161,161 0 0,1 417,256" fill="none" />
        <path id="rc-bot-arc" d="M 115,256 A 141,141 0 0,0 397,256" fill="none" />
        <path id="rc-ribbon-arc" d="M 70,385 Q 256,435 442,385" fill="none" />
      </defs>
      <circle cx="256" cy="256" r="250" fill="#F7EBDC" />
      <circle cx="256" cy="256" r="242" fill="#1F1009" />
      <circle cx="256" cy="256" r="230" fill="none" stroke="#F7EBDC" stroke-width="8" />
      <circle cx="256" cy="256" r="218" fill="none" stroke="#D9825B" stroke-width="3" />
      <text fill="#F7EBDC" font-size="56" font-weight="900" font-family="Georgia, serif" letter-spacing="14">
        <textPath href="#rc-top-arc" start-offset="50%" text-anchor="middle">RADHA</textPath>
      </text>
      <text fill="#D9825B" font-size="22" font-weight="800" font-family="sans-serif" letter-spacing="6">
        <textPath href="#rc-bot-arc" start-offset="50%" text-anchor="middle">ESTD  2026</textPath>
      </text>
      <circle cx="256" cy="236" r="142" fill="#543118" />
      <circle cx="256" cy="236" r="136" fill="none" stroke="#F7EBDC" stroke-width="4" />
      <circle cx="256" cy="236" r="128" fill="none" stroke="#D9825B" stroke-width="2" opacity="0.8" />
      <clipPath id="rc-medal-clip"><circle cx="256" cy="236" r="126" /></clipPath>
      <g clip-path="url(#rc-medal-clip)" stroke="#F7EBDC" stroke-width="1.5" opacity="0.25">
        <line x1="110" y1="150" x2="402" y2="150" />
        <line x1="110" y1="180" x2="402" y2="180" />
        <line x1="110" y1="210" x2="402" y2="210" />
        <line x1="110" y1="240" x2="402" y2="240" />
        <line x1="110" y1="270" x2="402" y2="270" />
        <line x1="110" y1="300" x2="402" y2="300" />
      </g>
      <g fill="#F7EBDC" opacity="0.95">
        <path d="M 234,180 C 220,150 260,130 276,105 C 282,96 284,85 277,75 C 272,92 244,110 234,130 C 223,150 230,165 234,180 Z" />
        <path d="M 264,178 C 250,152 284,134 300,112 C 306,104 310,95 304,85 C 299,99 272,118 264,134 C 253,152 261,165 264,178 Z" />
        <path d="M 216,190 C 206,168 234,154 245,135 C 250,127 252,120 247,112 C 243,123 226,137 219,149 C 210,165 215,180 216,190 Z" />
      </g>
      <ellipse cx="248" cy="192" rx="66" ry="14" fill="#F7EBDC" stroke="#1F1009" stroke-width="2.5" />
      <ellipse cx="248" cy="193" rx="58" ry="10" fill="#1F1009" />
      <path d="M 182,192 L 192,278 C 195,292 214,304 248,304 C 282,304 301,292 304,278 L 314,192 Z" fill="#F7EBDC" />
      <path d="M 182,192 L 192,278 C 195,292 214,304 248,304 C 282,304 301,292 304,192" fill="none" stroke="#F7EBDC" stroke-width="3.5" />
      <path d="M 312,208 C 350,212 358,258 306,278" fill="none" stroke="#F7EBDC" stroke-width="12" stroke-linecap="round" />
      <path d="M 312,208 C 350,212 358,258 306,278" fill="none" stroke="#543118" stroke-width="5" stroke-linecap="round" />
      <ellipse cx="248" cy="302" rx="78" ry="16" fill="#F7EBDC" stroke="#1F1009" stroke-width="2.5" />
      <ellipse cx="248" cy="312" rx="92" ry="18" fill="#F7EBDC" stroke="#1F1009" stroke-width="2.5" />
      <path d="M 292,312 C 322,302 356,290 372,276 C 380,288 366,312 338,324 C 320,330 298,332 276,328 Z" fill="#F7EBDC" opacity="0.9" />
      <g transform="translate(118, 240)">
        <ellipse cx="0" cy="0" rx="16" ry="25" fill="#F7EBDC" stroke="#1F1009" stroke-width="1.5" />
        <path d="M -1,-20 C 7,-9 -7,9 1,20" fill="none" stroke="#1F1009" stroke-width="3" stroke-linecap="round" />
      </g>
      <g transform="translate(394, 240)">
        <ellipse cx="0" cy="0" rx="16" ry="25" fill="#F7EBDC" stroke="#1F1009" stroke-width="1.5" />
        <path d="M -1,-20 C 7,-9 -7,9 1,20" fill="none" stroke="#1F1009" stroke-width="3" stroke-linecap="round" />
      </g>
      <path d="M 45,365 L 88,340 L 88,395 L 45,418 L 64,390 Z" fill="#C8B299" stroke="#1F1009" stroke-width="3" />
      <path d="M 467,365 L 424,340 L 424,395 L 467,418 L 448,390 Z" fill="#C8B299" stroke="#1F1009" stroke-width="3" />
      <path d="M 55,350 Q 256,392 457,350 L 475,398 Q 256,445 37,398 Z" fill="#EDE0D0" stroke="#1F1009" stroke-width="3.5" />
      <text fill="#1F1009" font-size="42" font-weight="900" font-family="sans-serif" letter-spacing="6">
        <textPath href="#rc-ribbon-arc" start-offset="50%" text-anchor="middle">COFFEE CAFE</textPath>
      </text>
    </svg>
  `;

  // 1. Generate 1200x630 Open Graph Banner
  await page.setViewportSize({ width: 1200, height: 630 });
  const ogHtml = `
    <!DOCTYPE html>
    <html>
    <head>
      <style>
        * { box-sizing: border-box; margin: 0; padding: 0; }
        body {
          width: 1200px;
          height: 630px;
          background: radial-gradient(circle at 60% 40%, #3E2318 0%, #170B06 70%, #0D0503 100%);
          display: flex;
          align-items: center;
          justify-content: space-between;
          padding: 60px 80px;
          font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, sans-serif;
          color: #F7EBDC;
          overflow: hidden;
          position: relative;
        }
        body::before {
          content: '';
          position: absolute;
          inset: 0;
          background: radial-gradient(circle at 20% 80%, rgba(217, 130, 91, 0.15), transparent 40%),
                      radial-gradient(circle at 80% 20%, rgba(247, 235, 220, 0.1), transparent 50%);
          pointer-events: none;
        }
        .content {
          max-width: 620px;
          z-index: 2;
        }
        .badge {
          display: inline-block;
          padding: 8px 18px;
          background: rgba(217, 130, 91, 0.18);
          border: 1px solid rgba(217, 130, 91, 0.4);
          border-radius: 999px;
          color: #D9825B;
          font-size: 16px;
          font-weight: 700;
          letter-spacing: 2px;
          text-transform: uppercase;
          margin-bottom: 24px;
        }
        h1 {
          font-size: 64px;
          font-weight: 900;
          line-height: 1.1;
          color: #F7EBDC;
          margin-bottom: 16px;
          letter-spacing: -1px;
        }
        h1 span {
          color: #D9825B;
        }
        p {
          font-size: 24px;
          color: rgba(247, 235, 220, 0.85);
          line-height: 1.4;
          margin-bottom: 32px;
          font-weight: 400;
        }
        .features {
          display: flex;
          gap: 16px;
        }
        .feature-pill {
          padding: 8px 16px;
          background: rgba(255, 255, 255, 0.07);
          border: 1px solid rgba(255, 255, 255, 0.12);
          border-radius: 10px;
          font-size: 15px;
          font-weight: 600;
          color: #F7EBDC;
        }
        .logo-wrap {
          width: 380px;
          height: 380px;
          filter: drop-shadow(0 25px 35px rgba(0,0,0,0.6));
          z-index: 2;
        }
      </style>
    </head>
    <body>
      <div class="content">
        <div class="badge">☕ Artisanal Cafe & POS</div>
        <h1>Radha<span>Cafe</span></h1>
        <p>A modern cafe management, digital billing, real-time analytics & thermal printing system.</p>
        <div class="features">
          <div class="feature-pill">⚡ Digital Billing</div>
          <div class="feature-pill">📊 Live Analytics</div>
          <div class="feature-pill">🖨️ Thermal Printing</div>
        </div>
      </div>
      <div class="logo-wrap">
        ${svgContent}
      </div>
    </body>
    </html>
  `;

  await page.setContent(ogHtml);
  await page.screenshot({ path: path.join(process.cwd(), 'public', 'og-image.png'), type: 'png' });
  console.log('Created public/og-image.png (1200x630)');

  // 2. Generate 512x512 Square Logo PNG
  await page.setViewportSize({ width: 512, height: 512 });
  const logoHtml = `
    <!DOCTYPE html>
    <html>
    <head>
      <style>
        * { box-sizing: border-box; margin: 0; padding: 0; }
        body {
          width: 512px;
          height: 512px;
          background: #170B06;
          display: flex;
          align-items: center;
          justify-content: center;
          padding: 24px;
        }
        .logo { width: 464px; height: 464px; }
      </style>
    </head>
    <body>
      <div class="logo">${svgContent}</div>
    </body>
    </html>
  `;

  await page.setContent(logoHtml);
  await page.screenshot({ path: path.join(process.cwd(), 'public', 'logo.png'), type: 'png' });
  await page.screenshot({ path: path.join(process.cwd(), 'public', 'apple-touch-icon.png'), type: 'png' });
  await page.screenshot({ path: path.join(process.cwd(), 'public', 'favicon.png'), type: 'png' });
  console.log('Created public/logo.png, public/apple-touch-icon.png, public/favicon.png (512x512)');

  await browser.close();
}

generateImages().catch(console.error);
